src/db_options.rs

```diff
@@ -130,6 +130,7 @@ pub struct BlockBasedOptions {
 
 pub struct ReadOptions {
     pub(crate) inner: *mut ffi::rocksdb_readoptions_t,
+    iterate_upper_bound: Option<Vec<u8>>,
```

Comment:
I would suggest using `Option<Box<[u8]>>` instead. First, it occupies less space (16 bytes vs 24 bytes) and second, it guarantees the pointer will remain the same (vec can reallocate, it doesn't in this case, but still it's not guaranteed at compile-time).